helper/ssh/ssh.go

```diff
@@ -42,3 +51,62 @@ func FileSigner(path string) (ssh.Signer, error) {
 
 	return signer, nil
 }
+
+// FileSigner returns an ssh.Signer for a key file.
+func FileSignerWithCert(path string, certificatePath string) (ssh.Signer, error) {
+
+	keySigner, err := FileSigner(path)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if certificatePath == "" {
+		return keySigner, err
+	}
+
+	// Load private key
+	if err != nil {
+		return nil, fmt.Errorf("unable to read key file: %v", err)
+	}
+
+	// Load the certificate
+	cert, err := ioutil.ReadFile(certificatePath)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read certificate file: %v", err)
+	}
+
+	pk, _, _, _, err := ssh.ParseAuthorizedKey(cert)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse public key: %v", err)
+	}
+
+	certificate := pk.(*ssh.Certificate)
```

Comment:
If this assertion fails, Packer will crash. Let's do it as a check instead: 
```suggestion
	certificate, ok := pk.(*ssh.Certificate)
	if !ok {
       	return nil, fmt.Errorf("Error loading certificate")
	}
```

Reply:
Fixed